cmd/blockchaincmd/change_weight.go

```diff
@@ -141,59 +134,133 @@ func setWeight(_ *cobra.Command, args []string) error {
 		return fmt.Errorf("node %s is not a validator for blockchain %s", nodeID, subnetID)
 	}
 
-	deployer := subnet.NewPublicDeployer(app, kc, network)
+	chainSpec := contract.ChainSpec{
+		BlockchainName: blockchainName,
+	}
 
-	// first remove the validator from subnet
-	err = removeValidatorSOV(deployer,
-		network,
-		blockchainName,
-		nodeID,
-		0, // automatic uptime
-		isBootstrapValidatorForNetwork(nodeID, sc.Networks[network.Name()]),
-		false, // don't force
-	)
+	if rpcURL == "" {
+		rpcURL, _, err = contract.GetBlockchainEndpoints(
+			app,
+			network,
+			chainSpec,
+			true,
+			false,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	validationID, err := validatorManagerSDK.GetValidationID(rpcURL, nodeID)
 	if err != nil {
 		return err
 	}
 
-	// TODO: we need to wait for the balance from the removed validator to arrive in changeAddr
-	// set arbitrary time.sleep here?
-
-	weight, err = app.Prompt.CaptureWeight("What weight would you like to assign to the validator?")
+	vdrInfo, err := validatorManagerSDK.GetValidatorInfo(network.SDKNetwork(), validationID)
```

Comment:
sorry not happy with variable name `validatorInfo` in much better name imho
pls feel free ignore this comment :)